source/main.cc

```diff
@@ -579,6 +582,11 @@ int main (int argc, char *argv[])
           use_threads = true;
 #endif
         }
+      else if (arg == "--test")
+        {
+          run_unittests = true;
+          break;
+        }
```

Comment:
Might as well run the tests right here and return, rather than defer to below.

Reply:
I would like to do this after initializing MPI so we can later create vectors, etc..

Reply:
Ah, I see. OK then.